src/Chunk.ts

```diff
@@ -1131,11 +1157,13 @@ export default class Chunk {
 					exportVariable.setRenderNames('exports', exportName);
 				} else {
 					exportVariable.setRenderNames(null, null);
+					if (options.format === 'es' && exportVariable instanceof SyntheticNamedExportVariable) {
+						exportVariable.setSafeName(getSafeName(exportVariable.name, usedNames));
```

Comment:
Even though it would cover most cases to do it here, deconflicting in this place would not prevent conflicts with global variables of the same name. I added a test for this. As a rule of thumb, only format specific globals should be handled in this method as we cannot "fix" it if the user accesses them. All other deconflicting should happen in `deconflictChunk.ts`. I think it should be easy to fix this by deconflicting the synthetic named exports for each module in `deconflictTopLevelVariables` in that file.

Reply:
managed to fix this one! thanks for the tip